npm-build-push/action.yaml

```
@@ -0,0 +1,50 @@
+name: Npm build and push
+description: Build and Push npm package to registry
+
+inputs:
+  version:
+    description: "Node version to setup. Set to empty to not run a setup. Default: 18.x"
+    default: "18.x"
+  packages:
+    description: "Space separated string of packages to build and push. If not set no package will build. Default: empty"
+  registry-url:
+    description: "The registry url used to push npm packages to. Default: https://npm.pkg.github.com"
+    default: "https://npm.pkg.github.com"
+  registry-token:
+    description: "Registry login password/token. If not set packages will not pushed to registry. Default: empty"
+
+branding:
+  icon: "package"
+  color: "green"
+
+runs:
+  using: "composite"
+  steps:
+    - name: Setup Node.js
+      if: inputs.version
+      uses: actions/setup-node@v3
+      with:
+          node-version: ${{ inputs.version }}
+          registry-url: ${{ inputs.registry-url }}
+
+    - name: Run npm ci
+      shell: bash
+      run: npm ci
```

```suggestion
    - name: Run npm ci
      shell: bash
      run: npm c
      if: ${{ inputs.run-ci }}
```

I do not think, that the CI is required for this step? Maybe you could make this an optional step?

---

done